src/components/Trade/hooks/useSwapper/useSwapper.ts

```diff
@@ -234,18 +254,43 @@ export const useSwapper = () => {
     if (!swapper) throw new Error('no swapper available')
     if (!wallet) throw new Error('no wallet available')
 
+    const { chainId: receiveAddressChainId } = fromAssetId(buyAsset.assetId)
+    const chainAdapter = getChainAdapters().get(receiveAddressChainId)
+
+    if (!chainAdapter) throw new Error(`couldnt get chain adapter for ${receiveAddressChainId}`)
+
+    const receiveAddressAccountSpecifiers = accountSpecifiersList.find(
+      specifiers => specifiers[buyAsset.chainId],
+    )
+
+    if (!receiveAddressAccountSpecifiers) throw new Error('no receiveAddressAccountSpecifiers')
+    const receiveAddressAccountSpecifier = receiveAddressAccountSpecifiers[buyAsset.chainId]
+    if (!receiveAddressAccountSpecifier) throw new Error('no receiveAddressAccountSpecifier')
+
+    const { accountType: receiveAddressAccountType, utxoParams: receiveAddressUtxoParams } =
+      accountIdToUtxoParams(receiveAddressAccountSpecifiers[buyAsset.chainId], 0)
+
+    const receiveAddress = await chainAdapter.getAddress({
+      wallet,
+      accountType: receiveAddressAccountType,
+      ...receiveAddressUtxoParams,
+    })
+
     const tradeQuote = await (async () => {
-      if (sellAsset.chainId === KnownChainIds.EthereumMainnet) {
+      if (supportedSwappingChain(sellAsset.chainId)) {
         return swapper.buildTrade({
-          chainId: sellAsset.chainId,
+          chainId: sellAsset.chainId as
```

Comment:
If you make `supportedSwappingChain(sellAsset.chainId` a type guard that returns `is | KnownChainIds.EthereumMainnet | KnownChainIds.CosmosMainnet | KnownChainIds.OsmosisMainnet`, `chainId` will be typed properly here and you won't have to do any casting